arraycontext/impl/jax/fake_numpy.py

```diff
@@ -62,7 +63,11 @@ def full_like(self, ary, fill_value):
         def _full_like(subary):
             return jnp.full_like(ary, fill_value)
 
-        return self._new_like(ary, _full_like)
+        if isinstance(ary, np.ndarray) and ary.dtype.char == "O":
+            # NOTE: see `arraycontext.fake_numpy.empty_like` for comments
+            raise NotImplementedError("operation not implemented for object arrays")
```

Comment:
This is weird now that object arrays are array containers, and array containers are supported...

Reply:
I'm leaning towards allowing this now
* it seems useful to do things like `actx.np.ones_like(nodes)` or `actx.np.ones_like(velocity)`
* we would need to be consistent with `zeros_like` and `empty_like` too.

What do you think?

Reply:
Yeah, I agree that this should be legal.

Reply:
Removed the check in https://github.com/inducer/arraycontext/compare/2dcdbfe4f19d33f96fa92f376a558ebd98cc1780..838fc71c77131b7759702ea78b5178bfd9ca2fb2.